### DISCUSSION TITLE
About usage of each port for Hubs Cloud?

### DISCUSSION THREAD

**Discussion by jasoncks** [mozilla/hubs#3288] [Hubs Cloud Questions] [answered]
We find that in [troubleshooting](https://hubs.mozilla.com/docs/hubs-troubleshooting.html) page, below information is added.

> "Unable to connect" error
An error that says "unable to connect" is often caused by a firewall. At a minimum, Hubs needs your computer to connect to external ports 80 and 443, both via TLS. Ideally, you should also open up ports 51610-65535 on UDP and TCP for Hubs on your router's firewall.
>Note that for self-hosted versions of Hubs you need to open ports 49152-60999 on UDP and TCP. You will also need to open port 19302 on UDP and TCP.

Since we need to explain to our client why the ports are needed to be opened.
May you provide more information about the usage of each port?

If possible, can we have the information as below table.

Port No | Usage
-- | --
80 |
443 |
19302 |  
49152-60999 |  
51610-65535 |  

Thank you.

**Comment by vincentfretin** [comment, marked as answer]
From my understanding of what I read in various places and on https://github.com/mozilla/hubs-ops
- 80 tls to connect to the TURN server (coturn)
- 443 tls (https) to connect to the reticulum backend
- 19302 : Google STUN server
- 49152-60999 port range used for WebRTC audio/video: 49152-51609 for coturn I think and 51610-60999 or 51610-65535 for Dialog mediasoup WebRTC server

**Comment by jasoncks** [comment]
Just want to confirm one more things.
I am using Hubs Cloud AWS with custom client.

I found that the inbound/outbound rules setting does not actually include Port 19302, 49152-51609.
But included 5349, 51610 - 60999.

Is that meas Port 19302, 49152-51609 are not used in Hubs Cloud AWS.
And what is the use of port 5349?

**Reply by vincentfretin** [reply]
19302 is for stun:stun1.l.google.com:19302 it should be open on your router, it's your browser that try to connect to it. You can look in chrome with chrome://webrtc-internals/ you will see for example turns:lucid-ardent.reticulum.io:80, turns:lucid-ardent.reticulum.io:80?transport=tcp, stun:stun1.l.google.com:19302

And I know the Google STUN server was used by janus on Hubs Cloud as well because it's behind a NAT https://github.com/mozilla/hubs-ops/blob/73a82208d071baf3aa692b737c6f863f01c149e9/ansible/roles/janus/templates/janus-gateway.toml.j2#L15-L16
Now that Hubs Cloud switched to Dialog mediasoup, I guess it's still the case, but I couldn't find in the code where it's defined.

So you see in chrome://webrtc-internals/ the server uses turns, so it's TLS, on port 80. From https://github.com/mozilla/hubs-ops/blob/db2488f688f3bed50c6327cf2ee895c02f701210/terraform/modules/janus/main.tf#L227-L229
port 80 is redirected to coturn listening port 5349

For the 49152-51609 that is for the TURN server coturn, The TURN server may need to be enabled in the admin interface?
I never created a Hubs Cloud myself, so never saw the admin interface :) I just read the code, the documentation and conversations here.

**Reply by robinkwilson** [reply]
In parts of the code, we still have some variables called "janus" for dialog to ensure we didn't break anything during the migration from janus to dialog.

**Comment by dandrus104** [comment]
Can anyone answer this along the same lines? what I need is the destination IPS.  meaning which servers are we talking to on those ports?

**Reply by emclaren** [reply]
Hi @dandrus104  - you might have more success getting an answer if you create a new question (please add a link to this discussion for folks' reference), as this one has been tagged as "answered" and I'm not sure if the dev team will see this follow up question
